pkg/receive/handler_test.go

```diff
@@ -690,6 +690,87 @@ func TestReceiveQuorum(t *testing.T) {
 	}
 }
 
+func TestReceiveWriteRequestLimits(t *testing.T) {
```

Comment:
Maybe an e2e test could help in understanding some real-world client behavior eg Prometheus, avalanche. WDYT?

Reply:
Will add 👍

Reply:
@saswatamcode actually, what do you mean with `understanding some real-world client behavior eg Prometheus, avalanche.`? 

Do you mean, for example, testing that a well configured Prometheus or Avalanche will retry on a 429? I don't believe we need an e2e test to cover this.

I started working on the e2e tests and the further I progress on them the more I think they have no real value beyond what the handler test provides. The only additional value the e2e adds that I can see is to ensure that the CLI args are properly passed down to the request limiting code. Besides this, the request limiting code can be tested without having to add an e2e test.

Reply:
I meant if seeing how a hashring would behave, with such request limits and multiple tenants, would be useful?
But yes let's see what maintainers have to say! Unit tests might be sufficient here. 🙂

Reply:
Less + simpler tests while having the same verifications level = better, so I think a unit tests will suffice here. 

> It means we're testing Prometheus/Avalanche and not Thanos.

In some way yes, but also... are we sure those projects have those tests? 🙃 

Anyway, let's investigate/create focus tests in separate stream of work